pkgs/development/tools/pnpm/generic.nix

```diff
@@ -75,6 +79,28 @@ stdenvNoCC.mkDerivation (finalAttrs: {
       tests.version = lib.optionalAttrs withNode (
         testers.testVersion { package = finalAttrs.finalPackage; }
       );
+      updateScript = writeScript "discord-update-script" ''
+        #!/usr/bin/env nix-shell
+        #!nix-shell -i bash -p curl jq common-updater-scripts
+        set -eou pipefail;
+
+        curl_github() {
+            curl -L ''${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} "$@"
+        }
+
+        majorVersion=""
+
+        latestTag=$(curl_github https://api.github.com/repos/pnpm/pnpm/releases?per_page=100 | jq -r --arg major "${majorVersion}" '[.[].tag_name | select(startswith("v" + $major))][0]')
```

Comment:
```suggestion
        latestTag=$(curl -L ''${GITHUB_TOKEN:+" -u \":$GITHUB_TOKEN\""} https://api.github.com/repos/pnpm/pnpm/releases?per_page=100 | jq -r --arg major "${majorVersion}" '[.[].tag_name | select(startswith("v" + $major))][0]')
```

Reply:
I think this makes it harder to read

Reply:
First of all majorVersion is unused. We also don't need to hide things in functions for no reason.